pkg/function/service.go

```diff
@@ -20,9 +45,14 @@ func (fs *Service) IsReady(context.Context, *emptypb.Empty) (*functionpb.ReadyRe
 	return &functionpb.ReadyResponse{Ready: true}, nil
 }
 
-// DoFn applies a function to each datum element
-func (fs *Service) DoFn(ctx context.Context, d *functionpb.Datum) (*functionpb.DatumList, error) {
-	messages, err := fs.Mapper.HandleDo(ctx, d.GetKey(), d.GetValue())
+// MapFn applies a function to each datum element
+func (fs *Service) MapFn(ctx context.Context, d *functionpb.Datum) (*functionpb.DatumList, error) {
+	var hd = handlerDatum{
+		value:     d.GetValue(),
```

Comment:
to make it consistent, let's also pass key in the Map also via the gRPC Metatata.